File: scripts/con-par/parallel.pl
#!/usr/bin/env perl

use v5.38;
use threads;

sub worker($id) {
    my $result = 0;
    for (my $i = 0; $i < 1_000_000; $i++) {
        $result += sqrt($i);
    }
    say "Task $id result: $result";
}

my @threads = map { threads->create(\&worker, $_) } (1..2);
$_->join for @threads;
print "All tasks done!\n";

File: scripts/con-par/parallel.py
#!/usr/bin/env python3

import math
import multiprocessing

def worker(id):
    result = 0
    for i in range(1_000_000):
        result += math.sqrt(i)
    print(f"Task {id} result: {result}")

if __name__ == "__main__":
    process1 = multiprocessing.Process(target=worker, args=(1,))
    process2 = multiprocessing.Process(target=worker, args=(2,))

    process1.start()
    process2.start()

    process1.join()
    process2.join()

    print("All tasks done!")
